### DISCUSSION TITLE
Custom orientation widget from mesh with color

### DISCUSSION THREAD

**Discussion by machur** [pyvista/pyvista#4771] [Q&A] [answered]
Hi, I've been trying to change the color of a custom axes widget. I've created it from a mesh using the code below. Self is a plotter that inherits from QtInteractor class:

```python
    def _add_axes_marker(self):
        mesh = pv.read(directory / "human.obj")

        mapper = pv.DataSetMapper(mesh)
        actor = pv.Actor(mapper=mapper)
        axes_widget = self.renderer.add_orientation_widget(actor, interactive=False, color="black")
        axes_widget.SetViewport((0, 0, 0.2, 0.2))
```

There is a "color" argument in add_orientation_widget method and according to the docs it should work for datasets. It doesn't work for me though, my mesh is always in the default color. I tried to pass different types of values as color-like argument with no effect.

![image](https://github.com/pyvista/pyvista/assets/4650515/f0a25383-0d24-4020-9216-94b7b838f734)

Would you please advice?

**Comment by tkoyama010** [comment]
cc @pyvista/pyvistaqt-admin

**Comment by darikg** [comment, marked as answer]
> There is a "color" argument in add_orientation_widget method and according to the docs it should work for datasets

you're passing an actor, not a dataset. I think you need to either pass the mesh directly to add_orientation_widget, or manually apply the color to the actor `actor.prop.color = 'black'` beforehand (if you look at the source, that's all that add_orientation_widget does)

**Reply by machur** [reply]
Thanks @darikg, it worked. I though it was called inside add_orientation_widget method, I didn't see that it was not called for actor instance.
